azkaban-web-server/src/main/java/azkaban/webapp/AzkabanWebServer.java

```
@@ -142,53 +134,50 @@
-   * initial server
-   */
-  public AzkabanWebServer() throws Exception {
-    this(null, loadConfigurationFromAzkabanHome());
-  }
-
   @Inject
-  public AzkabanWebServer(final Server server, final Props props) throws Exception {
+  public AzkabanWebServer(final Props props,
+      final Server server,
+      final ExecutorManager executorManager,
+      final ProjectManager projectManager,
+      final TriggerManager triggerManager,
+      final MetricsManager metricsManager,
+      final MetricRegistry metricRegistry,
+      final SessionCache sessionCache,
+      final UserManager userManager,
+      final ScheduleManager scheduleManager,
+      final VelocityEngine velocityEngine) {
     this.props = requireNonNull(props);
-    this.server = server;
-
-    this.velocityEngine = configureVelocityEngine(props.getBoolean(VELOCITY_DEV_MODE_PARAM, false));
-    this.sessionCache = new SessionCache(props);
-    this.userManager = loadUserManager(props);
-
-    // TODO remove hack. Move injection to constructor
-    this.executorManager = SERVICE_PROVIDER.getInstance(ExecutorManager.class);
-    this.projectManager = SERVICE_PROVIDER.getInstance(ProjectManager.class);
-    this.triggerManager = SERVICE_PROVIDER.getInstance(TriggerManager.class);
-    this.metricsManager = SERVICE_PROVIDER.getInstance(MetricsManager.class);
-    this.registry = SERVICE_PROVIDER.getInstance(MetricRegistry.class);
+    this.server = requireNonNull(server);
+    this.executorManager = requireNonNull(executorManager);
+    this.projectManager = requireNonNull(projectManager);
+    this.triggerManager = requireNonNull(triggerManager);
+    this.metricsManager = requireNonNull(metricsManager);
+    this.registry = requireNonNull(metricRegistry);
+    this.sessionCache = requireNonNull(sessionCache);
+    this.userManager = requireNonNull(userManager);
+    this.scheduleManager = requireNonNull(scheduleManager);
+    this.velocityEngine = requireNonNull(velocityEngine);
```

@reallocf 
short answer: yes.
long answer: This is not just about injection. If you are exposing public APIs or arguments, it is always better to state your requirements up front. This makes it fail fast and helps us figure out the problems earlier.